### DISCUSSION TITLE
Sensor integration

### DISCUSSION THREAD

**Discussion by nexus256** [joleys/niko-home-control-II#120] [Q&A]
Great job, and love the sensor for electricity.
Would it be possible to also include following pulse sensors?
- water
- gas

It seems that I'm also missing a switched outlet that is configured in NHC2.

**Comment by joleys** [comment]
Pulse modules are not in the current API documentation.  I've asked Niko if this is on their work list.

**Comment by WaterschootG** [comment]
Any update on the pulse modules?

**Comment by tijsverkoyen** [comment]
@nexus256 I think this will be working when the rewrite is released. If I'm not mistaken you have tested the rewrite already? Could you confirm?

**Comment by tijsverkoyen** [comment]
@nexus256 v3.0.0 was releases last Friday, did you have the chance to upgrade and confirm this?

**Comment by joleys** [comment]
The pulse modules are still not included in the NHC API.
Switched outlets should be available.

**Comment by sgi3200** [comment]
I have dropped few emails to Niko support regarding to pulse meter that could be added to MQTT broker (running at connected controller) but no feedback yet. Since it's not a part of MQTT (as mentioned by @joleys), there is no way how to get data from and to add it to HA integration.

**Comment by jonassmets** [comment]
Any way to add gas usage to HA true the NHC Smart hub? Data of electricity and gas is readable in NHC? Only electricity is visible in HA?

**Comment by tijsverkoyen** [comment]
@jonassmets I don't have Gas usage linked.
Can you provide me with a device list? See https://github.com/joleys/niko-home-control-II#i-need-to-see-the-device-list, if you don't want to share it here, you can mail it to me: `niko-ha [at] Verkoyen [dot] eu`.

With that information I should be able to add it, **if it is present in the data provided by Niko**.

**Comment by tijsverkoyen** [comment]
@jonassmets The data is not exposed by Niko. Feel free to contact them and link them to this issue.

The data that is related (to me) is shown below, an none of the are containing any values.

````json
        {
          "Uuid": "XXX",
          "Identifier": "XXX",
          "Model": "generic",
          "Online": "True",
          "Name": "energyhome",
          "Technology": "nikohomecontrol",
          "Type": "energyhome",
          "Traits": [],
          "Parameters": [
            {
...
            }
          ],
          "Properties": [
            {
              "WaterVolume": ""
            },
            {
...
            },
            {
              "GasCost": ""
            },
            {
              "GasVolume": ""
            },
            {
              "WaterCost": ""
            }
          ],
          "PropertyDefinitions": [
            {
              "WaterVolume": {
                "HasStatus": "false",
                "CanControl": "false",
                "Description": "Range(0.0000,999999.9990,0.0001)"
              }
            },
...
            {
              "GasCost": {
                "HasStatus": "false",
                "CanControl": "false",
                "Description": "Range(0.00000,999999999.00000,0.00001)"
              }
            },
            {
              "GasVolume": {
                "HasStatus": "false",
                "CanControl": "false",
                "Description": "Range(0.0000,999999.9990,0.0001)"
              }
            },
            {
              "WaterCost": {
                "HasStatus": "false",
                "CanControl": "false",
                "Description": "Range(0.00000,999999999.00000,0.00001)"
              }
            }
          ]
        }
````

There is a NHC Gasmeter, but there is no data available:

````json
        {
          "Uuid": "XXX",
          "Identifier": "XXX",
          "Model": "nhc",
          "Online": "True",
          "Name": "gasmeter",
          "Technology": "p1",
          "Type": "gasmeter",
          "Traits": [],
          "Parameters": [
            {
              "Segment": "Central"
            }
          ]
        },
````
But this has no properties, so we can't read any data.

So at this point: we can't do much. I have mailed to Niko also to check if we can read the data elsewhere.

**Reply by milansanders** [reply]
Unfortunately the documentation suggests the current API version doesn't disclose the data we want (No 'Status' and 'Logging' is not disclosed through the Hobby API).
I sent Niko an email, let's hope their IT department has some free space in their backlog :)
![Screenshot from 2024-12-28 12-46-39](https://github.com/user-attachments/assets/1ee2e8da-5d0e-42d3-a345-33223cca5b41)
![Screenshot from 2024-12-28 12-45-50](https://github.com/user-attachments/assets/5fb79580-7c54-41cc-83df-d6f9c5363802)

**Reply by tijsverkoyen** [reply]
The data is "available" since Niko v2.23. I will close this discussion in favor of [ould suggest to continue the discussion in](https://github.com/joleys/niko-home-control-II/discussions/247)
